### PR TITLE
Add transaction to IDBVersionChangeEvent

### LINKED ISSUE #20: `IdbVersionChangeEvent` should expose the current `IdbTransaction`
This event is fired during a transaction with mode `versionchange`, which should be available in `event.target.transaction`.

Although accessing the `IdbDatabase` allows us to create/delete object stores and set their indexes, it doesn't allow to access current stores. We need to access them via the `IdbTransaction`.

It would allow to change the indexes of the existing object stores, and to fix data in the stores or even migrate data from one store to the other.

### PR DESCRIPTION
**Pull request by kurtbuilds**
Right now, there's no way to access the current transaction from IdbVersionChangeEvent, meaning that any modifications to existing stores is not possible.

This adds a method to access the current transaction.

Example usage of the API is something like this:

```rust
    let mut db_req: OpenDbRequest = IdbDatabase::open_u32("my_db2", 2)?;
    db_req.set_on_upgrade_needed(Some(|evt: &IdbVersionChangeEvent| -> Result<(), JsValue> {
        let version = evt.old_version() as u32;
        if version < 1 {
            evt.db().create_object_store_with_params(
                EMPLOYEE_STORE,
                &IdbObjectStoreParameters::new().key_path(Some(&IdbKeyPath::str("id"))),
            )?;
        }
        if version < 2 {
            evt.transaction()
                // .transaction_on_one_with_mode(EMPLOYEE_STORE, IdbTransactionMode::Versionchange)?
                .object_store(EMPLOYEE_STORE)?
                .create_index_with_params(
                    "email",
                    &IdbKeyPath::str("email"),
                    &IdbIndexParameters::new().unique(true).multi_entry(true),
                )?;
        }
        Ok(())
    }));
```

This fixes #20 .

### CODE REVIEW

**Comment by Alorel** [comment]
Apologies, been neglecting this repo a bit. I'll get this sorted later today or tomorrow, thanks for the PR :heart:

File: src/idb_database/idb_version_change_event.rs [hunk 1]
```diff
@@ -24,9 +26,14 @@ impl IdbVersionChangeEvent {
             .expect("Failed to unwrap IdbOpenDbRequest result")
             .unchecked_into();
 
+        let req: IdbRequest = event.target().expect("Failed to unwrap IdbOpenDbRequest event target as IdbRequest")
```
Comment (Alorel) [inline src/idb_database/idb_version_change_event.rs:29]: No point going through `event.target()` again when you can just cast the existing `req`

Comment (Alorel) [reply src/idb_database/idb_version_change_event.rs]: Patched it up, I'll be doing some other refactors before publishing the release so I'm not too worried about the formatting.

**Comment by naiquevin** [comment]
I was looking for this but looks like the code has gone through some refactoring post this PR was merged. Can you please confirm?

**Comment by Alorel** [comment]
> I was looking for this but looks like the code has gone through some refactoring post this PR was merged. The current struct `VersionChangeEvent` doesn't provide a `transaction` method. And it's `.0` field is private so can't use it from outside as well. Can you please confirm?

@naiquevin This PR was from 2023 - yeah, the code's changed quite a bit.

**Pull request merged**